site/docs/v1/tech/lambdas/self-service-registration.adoc

```diff
@@ -0,0 +1,70 @@
+---
+layout: doc
+title: Self Service Registration Validation
+description: An overview of the self-service registration validation lambda
+navcategory: customization
+---
+
+:sectnumlevels: 0
+
+== Self Service Registration Validation lambda
+
+When you have an link:/docs/v1/tech/guides/advanced-registration-forms[Advanced Registration Form] you may add a lambda
+to perform additional validation of the form fields as the user completes steps in the form.
+
+When you create a new lambda using the FusionAuth UI we will provide you an empty function for you to implement. If you are using the API to create the lambda you will need to ensure your function has the following signature:
```

Comment:
```suggestion
When you create a new lambda using the FusionAuth UI, we will provide you an empty function for you to implement. If you are using the API to create the lambda, you will need to ensure your function has the following signature:
```

Reply:
Questioning this suggestion, not sure that's proper comma usage...

Reply:
`d,I` sentence structure for a [complex sentence](https://www.grammarly.com/blog/complex-sentence/?gclid=Cj0KCQiA_P6dBhD1ARIsAAGI7HAftvTK8G4h5UtZfW8j3Tn9RX1g847M-a6fO8PJ1JOhuwfTTamJvigaAkdIEALw_wcB&gclsrc=aw.ds).

The use of a subordinating conjunction (such as "when" or "if") creates a dependent clause. If the dependent clause in a complex sentence comes before the independent clause, a comma should follow the dependent clause.